include/revng/Yield/PTML.h

```diff
@@ -22,4 +22,7 @@ std::string controlFlowNode(const MetaAddress &BasicBlockAddress,
                             const yield::Function &Function,
                             const model::Binary &Binary);
 
+template<bool IsFake = false>
```

Comment:
Uh?

Reply:
This interface is kind of WIP, sorry. As of now, this "fake" value gets transposed onto the SVG as is. We need to decide on a proper way to encode that in our PTML (maybe after `@mrjackv` is back), then this interface will follow that.

Reply:
What's *fake* though? Fake functions? They no longer exists after `feature/rework-efa`.

Reply:
Sorry for not being clear. In call graphs we have "real" function nodes - those with the trees attached, and the "fake" ones - those that just serve as a "pointer" to the corresponding real node.

p. s. "fake" definitely wasn't a good name for this, was it :sweat_smile:.

Reply:
Ah OK, gotcha.

Reply:
I'm reopen this since we actually need to solve this. It's not acceptable as it stands. What I'm looking for is the official (tm) way to represent these "fake" nodes in the PTML. Maybe @mrjackv has something to say about it.

Reply:
If I understand the use-case correctly (pointing with locations with real and fake functions) I'd say to use a separate location (e.g. `/fakecfgfunction/0x1000`), I've done something similar for helper/synthetic structs in revng-c

Reply:
That sounds good for me. @aleclearmind what's your stance?

Reply:
Why not just a PTML *reference* (not definition) to `/function/0x1000`? From a semantic POV I think we can interpret fake functions as a reference to the actual one.
If I click on a *fake* function and it brings me to the non-fake version, I'm really happy.

Reply:
Everything is a reference. There are no definitions in this artifact - since it doesn't introduce anything. All it does is references a bunch of stuff with cool looking arrows in-between.

Reply:
I think the non-fake nodes can be definitions. So you can navigate to their location in the call graph.

Reply:
> I think the non-fake nodes can be definitions. So you can navigate to their location in the call graph.

I totally second this!

Reply:
Yeah, I agree about promoting some of the names to definitions, but I still think we can benefit from an explicit way to say "this part of the graph is cut out and you can find it if you click on this" other than implying that by making the node a reference.

Reply:
So what are you proposing?
Consider that in VSCode, if you right click on a reference, you can still click "Find references" and all the "sister" references pop up.

Reply:
I'm proposing both. To have one "definition" per function in the graph BUT also a way to mark some trees "Fake" independently of that. Maybe a custom `data-something` or the like.

Reply:
Unless you have a very compelling way to express this in VSCode, I'm in favor of the KISS logic.
PTML should be simple and obvious.

Reply:
My main point is that I don't want to link any fake-tree-specific css to something being a `data-reference`. What if we ever want to add some additional references to the graph? I feel like the fact that we cut out some part of the graph is standalone enough to have its own indication.

Reply:
Whoops, I forgot to close this one. It's done: https://github.com/revng/revng/pull/191/commits/85e5d60335c7e508c7f4202fb4d1e3292f78b997.